### PR TITLE
Add functions to read from Ostentus and move registers definitions to header file

### PR DESCRIPTION
**Pull request by szczys**
This PR adds the following functionality:
* Read single byte and byte array from Ostentus
* Read firmware version from Ostentus
* Read the number of available fifo slots from Ostentus
* Reset Ostentus

Additionally, the register map has been abstracted to a header file. This repo is now included as a submodule of the Ostentus repo, and the register map header file is used as the source or truth for both repos.

### CODE REVIEW

**Comment by cdwilson** [comment]
@szczys (not strictly related to this PR, but saw it while I was looking at the code just now)

Is this needed?

https://github.com/golioth/libostentus/blob/02a98295acb641b2ddc0df5fff63a33f2d3cdf1b/libostentus.c#L89-L95

i.e. `msgs[]` is always fixed at 3 messages (`reg`, `data1`, `data2`) and `num_msgs` is being set here:

https://github.com/golioth/libostentus/blob/02a98295acb641b2ddc0df5fff63a33f2d3cdf1b/libostentus.c#L79

**Review by cdwilson — COMMENTED** [review]
@szczys sorry for the delay on this... I wanted to actually test these changes out for a bit on the modbus reference design since it has the most slides of any RD.

Here's a link that shows the changes I made to test this PR with the modbus RD: https://github.com/golioth/reference-design-mb/pull/3/files

Couple things to note from that exercise:

I didn't make any changes to `libostentus` itself in this test (other than what's in this PR), but instead I created some wrapper/helper functions in [`ostentus.c`](https://github.com/golioth/reference-design-mb/blob/e95d07b9cd0fccaddb76caa4451629e0d79c75f5/src/ostentus.c) to get the API that I wanted for the app.

Since Ostentus will drop any messages sent while the FIFO is full, I wanted to check the FIFO's remaining slots before sending an I2C message. To do this, I added an [`ostentus_ready`](https://github.com/golioth/reference-design-mb/blob/e95d07b9cd0fccaddb76caa4451629e0d79c75f5/src/ostentus.c#L16-L41) helper function that returns `true`/`false` whether the FIFO is ready before the specified `timeout` expires. I added wrappers around each `libostentus` function in `ostentus.c` so that they now block waiting for Ostentus to be ready, for example:

https://github.com/golioth/reference-design-mb/blob/e95d07b9cd0fccaddb76caa4451629e0d79c75f5/src/ostentus.c#L141-L153

This let me keep the modbus app code basically the same as it was previously, but with the benefit that it will now automatically wait as-needed when the Ostentus FIFO is full (up to `OSTENTUS_READY_TIMEOUT` duration in the worst case):

https://github.com/golioth/reference-design-mb/blob/e95d07b9cd0fccaddb76caa4451629e0d79c75f5/src/main.c#L258-L301

While I'm currently doing all this in the app right now in `ostentus.c`, it would be nice if some of this functionality was available in `libostentus` so users of the lib could just call `slide_add` and not have to add any extra logic to check whether the FIFO is ready.

One idea would be to update the functions in `libostentus` so that they use `ostentus_ready`, i.e. something like this:

```c
int slide_add(uint8_t id, char *str, uint8_t len, k_timeout_t poll_interval, k_timeout_t timeout)
{

	if (!ostentus_ready(poll_interval, timeout)) {
		return -EBUSY;
	}

	return ostentus_i2c_write2(OSTENTUS_SLIDE_ADD, &id, 1, str, len);
}
```

That would give the user of libostentus the ability to control how long each call to Ostentus waits before giving up and the polling period (including no wait by specifying `K_NO_WAIT` as the timeout):

```
# polls every 50ms until 1s timeout expires
slide_add(TEMP_F, LABEL_TEMP, strlen(LABEL_TEMP), K_MSEC(50), K_MSEC(1000));

# sends the message immediately without waiting
slide_add(TEMP_F, LABEL_TEMP, strlen(LABEL_TEMP), K_NO_WAIT, K_NO_WAIT);
```

The downside of this approach is that the user of libostentus has to think about and specify the appropriate polling period and timeout to set for each call.

Another option (which is what I did in my app's `ostentus.h`), is just to define a fixed polling interval and timeout period:

```
#define OSTENTUS_READY_POLL_INTERVAL K_MSEC(50)
#define OSTENTUS_READY_TIMEOUT	     K_MSEC(1000)
```

This keeps the function signatures the same as the existing libostentus, but the library user looses the ability to control if/how long the functions block.

LMK what you think or if you have other ideas about how to implement something like this. In my ideal world, I'd like to just use libostentus in my apps without having to think about any of this FIFO stuff, but still take advantage of the `ostentus_fifo_ready` functionality "under the hood" so no messages are silently dropped by Ostentus.

p.s. in `ostentus.c` I prefixed all my helper functions with the `ostentus_` prefix. I think it might be good if we could use a prefix in libostentus to avoid any name collisions (since function names like `update_display` could show up in other libraries).

**Comment by cdwilson** [comment]
Another option if we wanted to update libostentus to use `ostentus_fifo_ready` but avoid anything that blocks waiting:

```
int slide_add(uint8_t id, char *str, uint8_t len)
{
	uint8_t slots_remaining;

	err = ostentus_fifo_ready(&slots_remaining);
	if (err) {
		LOG_ERR("Error reading Ostentus I2C FIFO remaining slots: %d", err);
		return err;
	}

	if (slots_remaining == 0) {
		return -EBUSY;
	}

	return ostentus_i2c_write2(OSTENTUS_SLIDE_ADD, &id, 1, str, len);
}
```

This would simply return an error in the case where the FIFO is full (instead of silently dropping the message on Ostentus), but wouldn't automatically wait for the FIFO to be ready (the library user would have to provide their own retry logic).

**Comment by ChrisGammell** [comment]
> @szczys sorry for the delay on this... I wanted to actually test these changes out for a bit on the modbus reference design since it has the most slides of any RD.
> 
> Here's a link that shows the changes I made to test this PR with the modbus RD: https://github.com/golioth/reference-design-mb/pull/3/files

On the above comments, if Mike hasn't already mentioned, could you push these into a new PR?

File: include/libostentus.h [hunk 1]
```diff
@@ -26,6 +27,9 @@ int slide_add(uint8_t id, char *str, uint8_t len);
 int slide_set(uint8_t id, char *str, uint8_t len);
 int summary_title(char *str, uint8_t len);
 int slideshow(uint32_t setting);
+int ostentus_version_get(char *buf, uint8_t buf_len);
```
[the remark this answers is not in the document]
Comment (szczys) [reply include/libostentus.h:30]: I like the specificity of indicating how the function is used in the name. Without the `_get` it's unclear if you're setting a value to match which version of the hardware is attached, or querying the version that is attached.

**Comment by szczys** [comment]
> @szczys (not strictly related to this PR, but saw it while I was looking at the code just now)
> 
> Is this needed?
> 
> https://github.com/golioth/libostentus/blob/02a98295acb641b2ddc0df5fff63a33f2d3cdf1b/libostentus.c#L89-L95
> 
> i.e. `msgs[]` is always fixed at 3 messages (`reg`, `data1`, `data2`) and `num_msgs` is being set here:
> 
> https://github.com/golioth/libostentus/blob/02a98295acb641b2ddc0df5fff63a33f2d3cdf1b/libostentus.c#L79

Please open a PR if this change is needed. It should be fixed (and tested) outside the scope of this PR.

**Pull request merged**